src/components/Skill/form.tsx

```diff
@@ -0,0 +1,155 @@
+import React, { useEffect, useState } from "react";
+import { Controller, useForm } from "react-hook-form";
+import { zodResolver } from "@hookform/resolvers/zod";
+import { z } from "zod";
+
+import { axiosInstance } from "@/lib/axiosinstance";
+import {
+  DialogHeader,
+  DialogTitle,
+  DialogDescription,
+  DialogFooter,
+} from "@/components/ui/dialog";
+import { Button } from "@/components/ui/button";
+import {
+  Select,
+  SelectTrigger,
+  SelectItem,
+  SelectValue,
+  SelectContent,
+} from "@/components/ui/select";
+import { Textarea } from "@/components/ui/textarea";
+
+interface SkillData {
+  skill: string;
+  description: string;
+}
+
+interface AddSkillFormProps {
+  newSkill: { id: string; skill: string; description: string };
+  setNewSkill: React.Dispatch<
+    React.SetStateAction<{ id: string; skill: string; description: string }>
+  >;
+  handleAddSkill: () => void;
+}
+
+const skillSchema = z.object({
+  skill: z.string().nonempty("Please enter a skill"),
+  description: z.string().nonempty("Please enter a description"),
+});
+
+const AddSkillForm: React.FC<AddSkillFormProps> = ({
+  newSkill,
+  setNewSkill,
+  handleAddSkill,
+}) => {
+  const {
+    control,
+    handleSubmit,
+    formState: { errors },
+    reset,
+  } = useForm<SkillData>({
+    resolver: zodResolver(skillSchema),
+    defaultValues: {
+      skill: "",
+      description: "",
+    },
+  });
+
+  const [skills, setSkills] = useState<{ label: string }[]>([]);
+
+  useEffect(() => {
+    async function fetchSkills() {
+      try {
+        const response = await axiosInstance.get("/skills/all");
+        setSkills(response.data.data);
+      } catch (error) {
+        console.error("Error fetching skills:", error);
+      }
+    }
+
+    fetchSkills();
+  }, []);
+
+  const onSubmit = async (data: SkillData) => {
+    try {
+      await axiosInstance.post(`/skills/createskill`, data);
```

Comment:
check response to see if skill was created or not
add a toast for both success and fail case